test/test-integration/src/test/java/org/wso2am/micro/gw/tests/util/Utils.java

```diff
@@ -0,0 +1,347 @@
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException ignored) {
+                }
+            } finally {
+                try {
+                    if ((socket != null) && (socket.isConnected())) {
+                        socket.close();
+                    }
+                } catch (IOException e) {
+                    log.error("Can not close the socket with is used to check the server status ", e);
+                }
+            }
+        }
+        throw new RuntimeException("Port " + port + " is not open");
+    }
+
+    /**
+     * wait until port is closed within given timeout value in mills.
+     *
+     * @param port    - port number
+     * @param timeout - mat time to wait
+     */
+    public static void waitForPortToClosed(int port, int timeout) {
+        long time = System.currentTimeMillis() + timeout;
+        boolean portOpen = Utils.isPortOpen(port);
+        while (portOpen && System.currentTimeMillis() < time) {
+            // wait until server shutdown is completed
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException ignored) {
+                //ignore
+            }
+            portOpen = Utils.isPortOpen(port);
+        }
+        if (portOpen) {
+            throw new RuntimeException("Port not closed properly when stopping server");
+        }
+    }
+
+    /**
+     * Check whether given port is in use or not.
+     *
+     * @param port - port number
+     * @throws MicroGWTestException if port is already in use
+     */
+    public static void checkPortAvailability(int port) throws MicroGWTestException {
+
+        //check whether http port is already occupied
+        if (isPortOpen(port)) {
+            throw new MicroGWTestException("Unable to start carbon server on port " +
+                    (port) + " : Port already in use");
+        }
+    }
+
+    /**
+     * Check whether the provided port is open.
+     *
+     * @param port The port that needs to be checked
+     * @return true if the <code>port</code> is open & false otherwise
+     */
+    public static boolean isPortOpen(int port) {
+        Socket socket = null;
+        boolean isPortOpen = false;
+        try {
+            InetAddress address = InetAddress.getLocalHost();
+            socket = new Socket(address, port);
+            isPortOpen = socket.isConnected();
+            if (isPortOpen) {
+                log.info("Successfully connected to the server on port " + port);
+            }
+        } catch (IOException e) {
+            log.info("Port " + port + " is closed and available for use");
+            isPortOpen = false;
+        } finally {
+            try {
+                if ((socket != null) && (socket.isConnected())) {
+                    socket.close();
+                }
+            } catch (IOException e) {
+                log.error("Can not close the socket with is used to check the server status ", e);
```

Comment:
Shall we rephrase is as 'Cannot close the socket with is used to check the server status'?

Reply:
Fixed in 3b6cc6755810d748cd3de4fb25696a5692c08e78